catalog/registry/catalog-registry-federationadminserviceimpl/src/test/java/ddf/catalog/registry/federationadmin/service/impl/FederationAdminServiceImplTest.java

```diff
@@ -0,0 +1,296 @@
+                "PRIMARY");
+        findThisMetacard.setAttribute(regNodeStatus);
+        QueryRequest request = getTestQueryRequest();
+        QueryResponse response = getPopulatedTestQueryResponse(request, findThisMetacard);
+
+        when(security.getSystemSubject()).thenReturn(subject);
+        when(catalogFramework.query(any(QueryRequest.class))).thenReturn(response);
+        when(registryTransformer.transform(any(InputStream.class))).thenReturn(addThisMetacard);
+
+        fasi.init();
+
+        verify(fasi).getRegistryMetacards();
+        verify(fasi, never()).addLocalEntry(addThisMetacard);
+    }
+
+    @Test
+    public void initWithQueryException() throws Exception {
+        when(security.getSystemSubject()).thenReturn(subject);
+        doThrow(SourceUnavailableException.class).when(catalogFramework)
+                .query(any(QueryRequest.class));
+
+        fasi.init();
+        verify(fasi).getRegistryMetacards();
+        verify(fasi, never()).addLocalEntry(any(Metacard.class));
+    }
+
+    @Test
+    public void initWithIngestException() throws Exception {
+        when(security.getSystemSubject()).thenReturn(subject);
+        doThrow(SourceUnavailableException.class).when(catalogFramework)
+                .create(any(CreateRequest.class));
+
+        fasi.init();
+        verify(fasi).getRegistryMetacards();
+        verify(fasi, never()).addLocalEntry(any(Metacard.class));
+    }
+
+    @Test
+    public void initWithMarshalException() throws Exception {
+        Metacard metacard = getTestMetacard();
+        when(security.getSystemSubject()).thenReturn(subject);
+        when(catalogFramework.query(any(QueryRequest.class))).thenReturn(null);
+        doThrow(ParserException.class).when(parser)
+                .marshal(any(ParserConfigurator.class), any(Object.class), any(OutputStream.class));
+
+        fasi.init();
+        verify(fasi).getRegistryMetacards();
+        verify(fasi, never()).addLocalEntry(any(Metacard.class));
+    }
+
+    @Test(expected = IngestException.class)
+    public void testAddLocalEntryCreateException() throws Exception {
+        Metacard metacard = getTestMetacard();
+
+        when(security.getSystemSubject()).thenReturn(subject);
+        doThrow(IngestException.class).when(catalogFramework).create(any(CreateRequest.class));
+
+
+        fasi.addLocalEntry(metacard);
+        verify(fasi).addLocalEntry(any(Metacard.class));
+    }
+
+    @Test(expected = UnsupportedQueryException.class)
+    public void testGetRegistryMetacardsUnsupportedQueryException() throws Exception {
+        when(security.getSystemSubject()).thenReturn(subject);
+        doThrow(UnsupportedQueryException.class).when(catalogFramework).query(any(QueryRequest.class));
+
+
+        fasi.getRegistryMetacards();
+        verify(fasi).getRegistryMetacards();
+    }
+
+
+
+
+
+
+
+
+    private Metacard getTestMetacard() {
+        Metacard metacard = new MetacardImpl(new RegistryObjectMetacardType());
```

Comment:
Oh, yeah. Same. I expected to be modifying the metacard but didn't. Since it's in a test class and to avoid kicking off another bamboo build, I'll just leave it unless I have to make any other changes for this PR. I'll be making more changes in the same area in another ticket and remove it then.